lib/ansible/modules/cloud/vmware/vmware_host_kernel_manager.py

```diff
@@ -0,0 +1,221 @@
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    kernel_module_name: "tcpip4"
+    kernel_module_option: "ipv6=0"
+
+- name: Using cluster_name, configure vmw_psp_rr options
+  vmware_host_kernel_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: '{{ virtual_cluster_name }}'
+    kernel_module_name: "vmw_psp_rr"
+    kernel_module_option: "maxPathsPerDevice=2"
+'''
+
+RETURN = r'''
+results:
+    description:
+    - dict with information on what was changed, by ESXi host
+    returned: success
+    type: dict
+    sample: {
+    "results": {
+        "myhost01.example.com": {
+            "changed": true,
+            "configured_options": "ipv6=0",
+            "msg": "Options have been changed on the kernel module",
+            "original_options": "ipv6=1"
+        }
+    }
+}
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+from ansible.module_utils._text import to_native
+
+
+class VmwareKernelManager(PyVmomi):
+    def __init__(self, module):
+        self.module = module
+        super(VmwareKernelManager, self).__init__(module)
+        cluster_name = self.params.get('cluster_name', None)
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        self.kernel_module_name = self.params.get('kernel_module_name')
+        self.kernel_module_option = self.params.get('kernel_module_option')
+        self.results = {}
+
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find a host system that matches the specified criteria")
+
+    # find kernel module options for a given kmod_name. If the name is not right, this will throw an exception
+    def get_kernel_module_option(self, host, kmod_name):
+        host_kernel_manager = host.configManager.kernelModuleSystem
+
+        try:
+            configured_module_options = host_kernel_manager.QueryConfiguredModuleOptionString(self.kernel_module_name)
+        except vim.fault.NotFound as kernel_fault:
+            self.module.fail_json(msg="Failed to find kernel module on host '%s'. More information: %s" % (host.name, to_native(kernel_fault.msg)))
+
+        return configured_module_options
+
+    # configure the provided kernel module with the specified options
+    def apply_kernel_module_option(self, host, kmod_name, kmod_option):
+        host_kernel_manager = host.configManager.kernelModuleSystem
+
+        if host_kernel_manager:
+            try:
+                if not self.module.check_mode:
+                    host_kernel_manager.UpdateModuleOptionString(kmod_name, kmod_option)
+                    self.results[host.name]['changed'] = True
```

Comment:
```suggestion
                 self.results[host.name]['changed'] = True
```

Reply:
Want to report changed as True even if module is ran under `check_mode`

Reply:
Ok, that is good to know.

I'm going to actually remove line 141 above because I have the changed status being assigned inside check_host_configuration_state

Reply:
I think this one is going to be resolved in the next commit akasurde. We can touch base tomorrow on this one in particular.